### ISSUE TITLE
Xml Documentation on Methods shows difference in SyntaxTree structure with VS2017 IDE code analysis in background Vs explicitly triggered msbuild

### ISSUE THREAD

**Issue by msbasanth** [dotnet/roslyn#24733]
**Version Used**: 
Roslyn v2.6.0
VS2017 15.5.6

**Steps to Reproduce**:

1. I have an analyzer for reading the XML Documentation and analyzing it. We have this analyzer added as an analyzer reference to a project where we have methods documented with XML Documentation.
E.g.

    ```csharp
        /// <summary>
        /// API:Yes
        /// </summary>
        public void Method()
        {
        }
    ```

2. I use below code snippet to read XML Documentation.

```csharp
var leadingTrivia = methodDeclarationSyntaxNode.GetLeadingTrivia();
var singleLineDocumentation = leadingTrivia.FirstOrDefault(trivia => trivia.Kind() == SyntaxKind.SingleLineDocumentationCommentTrivia);
```

3.  This works fine when we do code analysis from VS2017 IDE which is triggered in the background.
The leading trivia list retrieved from the method declaration syntax shows 3 trivias.
![xmldocumentationleadingtrivialistworkingcase](https://user-images.githubusercontent.com/17777813/36025675-df4efca6-0db9-11e8-93c3-c167ca5809a0.PNG)

Analyzer identifies this as proper XML documentation as it uses SingleLineDocumentationCommentTrivia for identifying XML Documentation.

4. When we are building the solution using msbuild (VS2017 IDE Rebuild All) there is no SingleLineDocumentationTrivia, all are SingleLineCommentTrivia.
![xmldocumentationleadingtrivialistnotworkingcase](https://user-images.githubusercontent.com/17777813/36025761-385b0628-0dba-11e8-939e-2985579de8ab.PNG)

**Expected Behavior**: 
Both VS2017 IDE Code analysis and msbuild compilation which triggers Roslyn analyzers should give same SyntaxTree structure for LeadingTrivia.

Actually, we should get SingleLineDocumentationTrivia as we have XML documentation, not just a comment node.

**Actual Behavior**:
VS2017 IDE Shows proper SingleLineDocumentationTrivia when we have a proper XML documentation. But msbuild triggered analyzer instance shows different syntax tree where only SingleLineCommentTrivia observed.

Please let me know your comment on this difference in behavior which is affecting one of our XML Documentation related analyzer.

Thanks
Basanth

**Comment by sharwell** [comment]
XML documentation output must be enabled, or the compiler will just treat them like normal comments. See #10003, #2773.

We created StyleCop Analyzers rule [SA0001](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA0001.md) for this scenario. It explains the requirement and provides workarounds for problems people may experience when turning this feature on.

**Comment by sharwell** [comment]
Duplicate of #10003.
